### ISSUE TITLE
Consider merging UniformDequantizeOp and UniformQuantizeOp into ConvertOp

### ISSUE THREAD

**Issue by burmako** [openxla/stablehlo#1576]
Why do we need three ops to express conversion between different kinds of types if one op (ConvertOp) appear to be sufficient? Sure, for unknown scales and zero points we'll want three ops, but we don't support them yet.

**Comment by sdasgup3** [comment]
I agree with this proposal. In the current scenario, where we do not have support for unknown scales and zero-points, we do not need any special operation. We can re-evaluate the need for such additional ops on need basis. 
 
With that I would propose to do the merge sooner than later. Otherwise, it may impact the producers/consumers of StableHLO who might have already adopted the `uniformQuantize/UniformDequantize` ops.

**Comment by sdasgup3** [comment]
@abattery 
Thanks for the feedback! 

>> This simplification might block such efforts.

Can you please elaborate a bit on this?  I am trying to understand if the `stablehlo.convert` op can be used for those efforts. Is this because you plan to add additional arguments to `stablehlo.quantize/stablehlo.dequantize` ops?

**Comment by abattery** [comment]
Yes, in quantized training, scales and zps should be calculated on the fly, which means that ops for quantization/dequantization should be enough flexible to get such parameters in the graph level.

**Comment by rewu93** [comment]
`unform_dequantize` is also used to requantize a quantized tensor. According to the spec, `convert` is `Performs an element-wise conversion from one element type to another on operand tensor and produces a result tensor.` 
Not sure if using `convert` here is clean (operands and results have the same type).

**Comment by sdasgup3** [comment]
>> unform_dequantize is also used to requantize a quantized tensor
I assume you meant `stablehlo.uniform_quantize` can also be used to re-quantize a quantized tensor. 

From the [convert spec](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#convert), only the shape should be the same, element_type could be different.

**Comment by rewu93** [comment]
> > > unform_dequantize is also used to requantize a quantized tensor
> > > I assume you meant `stablehlo.uniform_quantize` can also be used to re-quantize a quantized tensor.
> 
> From the [convert spec](https://github.com/openxla/stablehlo/blob/main/docs/spec.md?rgh-link-date=2023-06-12T18%3A43%3A53Z#convert), only the shape should be the same, element_type could be different.

Yes. But if we requantize a qi8 to another qi8 with different scale/zero point (or different axis), element_type` will be the same.  Maybe I misunderstood `convert` to `cast`.

**Comment by sdasgup3** [comment]
>> Yes. But if we requantize a qi8 to another qi8 with different scale/zero point (or different axis), element_type will be the same.

As scales and zero-poinst are part of the element-type, they will differ as well.


As we are still revisiting the use of unknown-zero points in the context of drq, I propose to revisit this item after we have a plan for speccing DRQ.
